backend/packages/Upgrade/src/api/services/FeatureFlagService.ts

```diff
@@ -874,6 +884,11 @@ export class FeatureFlagService {
         ];
 
         const segments = await this.segmentService.getSegmentByIds(segmentIds);
+
+        if (segmentIds.length !== segments.length) {
```

Comment:
This will fail if `segmentIds` have duplicate values. Use `Set` to remove duplicate values.

Reply:
code updated